explorer/interpreter/type_checker.cpp

```diff
@@ -309,6 +309,21 @@ static auto ExpectCompleteType(SourceLocation source_loc,
          << "incomplete type `" << *type << "` used in " << context;
 }
 
+// Expect that a type is concrete. Issue a diagnostic if not.
+static auto ExpectConcreteType(SourceLocation source_loc,
```

Comment:
Great! Would that be useable for `VariableDefinition` as well?

Reply:
> Great! Would that be useable for `VariableDefinition` as well?

Sure once this PR is merged I will use it for VariableDefinition and move implementation from Interpreter to Type Checker

Reply:
Perfect. I meant it would be best to do this for this PR, that way we don't introduce duplicated code and have guaranteed identical logic on both the `Declaration` and `Definition` side. Does that make sense?